src/libraries/System.Private.CoreLib/src/System/Buffers/ArrayPoolEventSource.cs

```diff
@@ -21,8 +22,9 @@ internal enum BufferAllocatedReason : int
             PoolExhausted
         }
 
-        // The ArrayPoolEventSource GUID is {0866b2b8-5cef-5db9-2612-0c0ffd814a44}
-        private ArrayPoolEventSource() : base(new Guid(0x0866b2b8, 0x5cef, 0x5db9, 0x26, 0x12, 0x0c, 0x0f, 0xfd, 0x81, 0x4a, 0x44), "System.Buffers.ArrayPoolEventSource") { }
+        // Parameterized constructor to block initialization and ensure the EventSourceGenerator is creating the default constructor
+        // as you can't make a constructor partial.
+        private ArrayPoolEventSource(int _) { }
```

Comment:
I'm not understanding this.  Does this imply that without this, the C# compiler will add the default ctor to the semantic model prior to the generator running?  If so, that seems flawed to me.

Reply:
Is to cause a compile error if the generator doesn't run because there is no default constructor i.e. to make the following fail without the generator running: `ArrayPoolEventSource Log = new ArrayPoolEventSource();`; for other methods they can be marked as `partial` for similar effect, but C# doesn't allow constructors to be partial.

Reply:
Is mainly to prevent a regression; as otherwise it will still work without the generator but invoke the full reflection path

Reply:
And then it'll be trimmed away because no one is using it?

Reply:
Should be I believe?